File: examples/proto/php/task.php
<?php

require_once "vendor/autoload.php";

use Monarchic\AgentProtocol\V1\AgentRole;
use Monarchic\AgentProtocol\V1\Task;

$task = new Task();
$task->setVersion("v1");
$task->setTaskId("task-123");
$task->setRole(AgentRole::DEV);
$task->setGoal("Implement protocol");

echo $task, PHP_EOL;

File: examples/proto/cpp/task.cpp
#include "monarchic_agent_protocol.pb.h"
#include <iostream>

int main() {
  monarchic::agent_protocol::v1::Task task;
  task.set_version("v1");
  task.set_task_id("task-123");
  task.set_role(monarchic::agent_protocol::v1::DEV);
  task.set_goal("Implement protocol");

  std::cout << task.ShortDebugString() << std::endl;
  return 0;
}
